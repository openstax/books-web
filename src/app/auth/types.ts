export interface User {
  firstName: string;
}

export interface State {
  user: User | undefined;
  established: boolean;
}
