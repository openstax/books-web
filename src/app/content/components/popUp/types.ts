import { LinkedArchiveTree, LinkedArchiveTreeNode, LinkedArchiveTreeSection } from '../../types';

export type LocationFilters = Map<
  string,
  { section: LinkedArchiveTreeNode, children?: LinkedArchiveTreeSection[] }
>;

export type LocationFiltersWithChildren = Map<
  string,
  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
>;

export interface FiltersChange<T> {
  remove: T[];
  new: T[];
}
