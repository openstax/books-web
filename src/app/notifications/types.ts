import { ActionType } from 'typesafe-actions';
import * as actions from './actions';

export interface Message {
  id: string;
  html: string;
  dismissable: boolean;
  start_at: string | null;
  end_at: string | null;
  url_regex: string | null;
}

export interface AppMessageNotification {
  type: 'Notification/appMessage';
  payload: Message;
}

export type Messages = Message[];

export type AnyNotification = ActionType<Pick<typeof actions, 'updateAvailable' | 'acceptCookies' >>
  | AppMessageNotification;

export type State = AnyNotification[];
