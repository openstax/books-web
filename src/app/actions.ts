import { createStandardAction } from 'typesafe-actions';
import { Experiments } from '../helpers/pollUpdates';

export const receiveFeatureFlags = createStandardAction('app/receiveFeatureFlags')<string[]>();
export const receivePageFocus = createStandardAction('app/receivePageFocus')<boolean>();
export const receiveExperiments = createStandardAction('app/receiveExperiments')<Experiments>();
