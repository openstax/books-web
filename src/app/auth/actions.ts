import { createStandardAction } from 'typesafe-actions';
import { User } from './types';

export const receiveUser = createStandardAction('Auth/receiveUser')<User>();
export const receiveLoggedOut = createStandardAction('Auth/receiveLoggedOut')();
