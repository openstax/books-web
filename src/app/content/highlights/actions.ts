import { createStandardAction } from 'typesafe-actions';

export const enableHighlighting = createStandardAction('FeatureFlag/enableHighlighting')<void>();
