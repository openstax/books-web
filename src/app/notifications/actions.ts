import { createStandardAction } from 'typesafe-actions';
import { AnyNotification, Message } from './types';

export const updateAvailable = createStandardAction('Notification/updateAvailable')();
export const acceptCookies = createStandardAction('Notification/acceptCookies')();
export const addToast = createStandardAction('Notification/toasts/add')
  .map((toast: {messageKey: string, shouldAutoDismiss?: boolean}) => ({
    payload: {
      messageKey: toast.messageKey,
      shouldAutoDismiss: toast.shouldAutoDismiss === undefined || toast.shouldAutoDismiss,
      timestamp: Date.now(),
    },
  }));

export const receiveMessages = createStandardAction('Notification/receiveMessages')<Message[]>();

export const dismissNotification = createStandardAction('Notification/dismiss')<AnyNotification>();
