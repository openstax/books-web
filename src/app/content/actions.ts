import { createStandardAction } from 'typesafe-actions';
import * as highlightingActions from './highlights/actions';
import * as searchActions from './search/actions';
import { ArchivePage, Book, Params, State } from './types';

export const search = searchActions;
export const highlighting = highlightingActions;

export const openToc = createStandardAction('Content/openToc')<void>();
export const closeToc = createStandardAction('Content/closeToc')<void>();
export const resetToc = createStandardAction('Content/resetToc')<void>();

export const requestBook = createStandardAction('Content/requestBook')<Params['book']>();
export const receiveBook = createStandardAction('Content/receiveBook')<Book>();

export const requestPage = createStandardAction('Content/requestPage')<Params['page']>();
export const receivePage = createStandardAction('Content/receivePage')<
  ArchivePage & {references: State['references']}
>();

export const closeCallToActionPopup = createStandardAction('Content/closeCallToActionPopup')();
export const openCallToActionPopup = createStandardAction('Content/openCallToActionPopup')();
