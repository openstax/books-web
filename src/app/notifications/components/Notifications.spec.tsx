import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestStore from '../../../test/createTestStore';
import MessageProvider from '../../MessageProvider';
import { Store } from '../../types';
import { acceptCookies, updateAvailable } from '../actions';
import ConnectedNotifications from './Notifications';

describe('Notifications', () => {
  let store: Store;

  beforeEach(() => {
    store = createTestStore();
  });

  it('matches snapshot', () => {
    store.dispatch(updateAvailable());
    store.dispatch(acceptCookies());

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ConnectedNotifications />
      </MessageProvider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot for unknown notification', () => {
    store.getState().notifications.push({type: 'foobar'} as any);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ConnectedNotifications />
      </MessageProvider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
