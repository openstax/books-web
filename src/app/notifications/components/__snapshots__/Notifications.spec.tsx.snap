// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches snapshot 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #fff;
  background-color: #f36b32;
  font-weight: bold;
  border: none;
}

.c7:hover {
  background-color: #e96128;
}

.c7:active {
  background-color: #df571e;
}

.c6 {
  display: grid;
  grid-gap: 1rem;
}

.c2 {
  width: 100%;
}

.c4 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.6rem;
  line-height: 2.5rem;
  margin: 0;
  padding: 1rem;
}

.c4 a {
  color: #027EB5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: #027EB5 solid 0.02em;
}

.c4 a:hover {
  color: #0064A0;
}

.c1 {
  margin: 0.5rem;
  background-color: #fff;
  border-style: solid;
  border-color: #e5e5e5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-width: thin;
  box-shadow: 0 1rem 2rem 0 rgba(0,0,0,0.2);
  overflow: visible;
}

.c1 > .c5 {
  padding: 1rem;
  margin: 0;
}

.c3 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.6rem;
  line-height: 2.5rem;
  padding: 0.5rem 1rem;
  font-weight: bold;
  background-color: #fafafa;
  line-height: 4rem;
  margin-bottom: 1rem;
}

.c3 a {
  color: #027EB5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: #027EB5 solid 0.02em;
}

.c3 a:hover {
  color: #0064A0;
}

.c0 {
  z-index: 2;
  top: 0;
  right: 0;
  overflow: visible;
  position: fixed;
}

@media (max-width:73.125em) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:73.125em) {
  .c4 {
    padding: 0 1rem;
  }
}

@media (max-width:73.125em) {
  .c1 {
    max-width: 117rem;
    margin: 0 auto;
    box-shadow: none;
    border: none;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:73.125em) {
  .c3 {
    line-height: 1rem;
    margin: 0;
    background-color: initial;
  }
}

@media (max-width:73.125em) {
  .c0 {
    z-index: 1;
    position: -webkit-sticky;
    position: sticky;
    border-bottom: thin solid #e5e5e5;
    padding: 0 3.2rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Updates Available
      </div>
      <p
        className="c4"
      >
        This page needs to be reloaded.
      </p>
    </div>
    <div
      className="c5 c6"
    >
      <button
        className="c7"
        onClick={[Function]}
      >
        Reload
      </button>
    </div>
  </div>
</div>
`;
